### PR TITLE
DAOS-10233 pool: add fault injection for pool upgrade (#8821)

### PR DESCRIPTION
**Pull request by wangshilong**
Add fault injection to pool upgrading, container upgrading
will be aborted if fault injection is inserted, use this
to verify pool upgrading could be resumed upon pool restarting

Signed-off-by: Wang Shilong <shilong.wang@intel.com>

### CODE REVIEW

**Comment by wangshilong** [comment]
I changed codes a bit from master PR to kick off mpi requirement, master did not have problem because master included
DAOS-7734 build: Wrap MPI code in unified interface change which might be  too much to backport to 2.2

**Review by liw — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]
OK for me

File: src/tests/suite/SConscript [hunk 1]
```diff
@@ -107,7 +107,6 @@ def scons():
                     daos_aggregate_ec.c daos_degrade_ec.c
                     daos_extend_simple.c daos_obj_ec.c""")
 
-    newenv.AppendUnique(LIBPATH=["$BUILD_DIR/src/client/dfs"])
```
Comment (mchaarawi) [inline src/tests/suite/SConscript:110]: i see on the master PR that this was moved to earlier, but here in the 2.2 one it's just removed. can you please explain why?

Comment (wangshilong) [reply src/tests/suite/SConscript:110]: I thought it could inherit from parent Sconscript, see it was appended in src/tests/Sconscript modification.

Comment (mchaarawi) [reply src/tests/suite/SConscript:110]: ah ok i see. i guess i was confused why this change was made then only for 2.2 and not master. but ok we can unblock this. probably better to keep master in sync to make future cherrypicking to the branches easier.

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**